api/src/data/migrations/003.create-departments.ts

```diff
@@ -0,0 +1,13 @@
+import * as knex from "knex";
+
+export async function up(knex: knex.Knex) {
+    await knex.schema.createTable("departments", function(table) {
+        table.string("code", 8).primary().notNullable();
```

Comment:
I think you want a primary key `id` in all tables.
If the "code" is unique, then give it a unique constraint.

Reply:
The `primary()` makes it a key and therefore unique

Reply:
@datajohnson I'm aware, I just like the simplicity of all foreign key relationships being `_id` to `.id`.